### ISSUE TITLE
Soho popdown to provide a "clickOutside" event

### ISSUE THREAD

**Issue by edgarinfor** [infor-design/enterprise-ng#761]
**Is your feature request related to a problem? Please describe.**
This request is related to issue #760 where we are having popdowns closing when the lookup fields are opened.

In our case, we also have **custom lookup fields** which behave very similar to soho's but, don't use the same exact code.

![wfm_popdown_01](https://user-images.githubusercontent.com/39097500/76870871-b4e00f80-6840-11ea-838d-b4e4a309d701.png)

![wfm_popdown_02](https://user-images.githubusercontent.com/39097500/76870883-b9a4c380-6840-11ea-914c-bba495204f7c.png)

**Describe the solution you'd like**
The soho popdown directive should have a "clickOutside" event that can be combined with **keepOpen** to have more control over when to close or not the popdown even for other things other than lookup dialogs.

**Describe alternatives you've considered**
We came up with a workaround that consists on using **keepOpen** on the popdowns and tracking when any click is done outside the popdowns. We combine that event with checking if a dialog has been opened before or not and decide if it was open from within the popdown and so, we decide to close it or not. 

The workaround was done by extending the **SohoPopDownDirective** with our own and doing all the extra code there; however, this API should belong to the directive because we had to access the popdown internals (e.g, JQuery, JS).

**Additional context**
The non-angular popdown .\src\components\popdown\popdown.js controls the click event (line 335):
```javascript
      if (!self.settings.keepOpen) {
        $(document)
          .on('click.popdown', handleFocusOut)
          .on('focusin.popdown', handleFocusOut);
      }
```
The click.popdown event should also be added when keepOpen is set and extend the handleFocusOut so it can use keepOpen internally and dispatch the clickOutside as needed.

You can refer to our customized directive for reference: [workbrain.popdown.ts.zip](https://github.com/infor-design/enterprise-ng/files/4344211/workbrain.popdown.ts.zip)

**Comment by tmcconechy** [comment]
We will still have to consider if we are going to deprecate all this in favor of popdown. As we have a lot of duplicate logic in the 2 (3 actual components). So once we get to that will take a look into this.

**Comment by edgarinfor** [comment]
@tmcconechy , I tried to replace the **popdown** with the **tooltip + popover** combo but, as we discussed in the channel, I encountered some issues.

Perhaps, you need a base "popover" directive (class) and then build the popdown and tooltip on top so they can coexists together and avoid functional collisions.

PS: I've been looking into to the code to try to add the contribution but, the functionality requested doesn't exist in the JQuery version so, I need to figure that out.

**Comment by tmcconechy** [comment]
Yeah i think you would need to add it to the jQuery function , actually i think this fix would go in that repo anyways so moving it there... https://github.com/infor-design/enterprise
